adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -581,6 +581,32 @@ def _perceive_property_assertions(self) -> None:
                     property_,
                 )
 
+        # Properties derived from the role of the situation object in the action
+        for action in self._situation.actions:
+            action_description: ActionDescription = GAILA_PHASE_1_ONTOLOGY.required_action_description(
+                action.action_type, action.argument_roles_to_fillers.keys()
+            )
+            for role in action_description.frame.semantic_roles:  # e.g. AGENT
+                variable = action_description.frame.roles_to_variables[
+                    role
+                ]  # e.g. _PUT_AGENT
+                fillers = action.argument_roles_to_fillers[role]  # e.g. {Mom}
+                for property_ in action_description.asserted_properties[variable]:
+                    for situation_or_region in fillers:
+                        if isinstance(situation_or_region, SituationObject):
+                            perception_of_object = self._objects_to_perceptions[
+                                situation_or_region
+                            ]
+                        else:
```

Comment:
Drop a comment on the `else` that we are here propagating properties asserted on regions to their reference objects; make an issue for post-M3 for us to think carefully about whether this is really the right thing to do.